src/index.ts

```diff
@@ -1,192 +1,8 @@
-	FilledLinkToMediaField,
-	RelationField,
-	LinkField,
-	LinkToMediaField,
-	// Embed
-	OEmbedExtra,
-	PhotoOEmbed,
-	VideoOEmbed,
-	LinkOEmbed,
-	RichOEmbed,
-	AnyOEmbed,
-	EmbedField,
-	// Simple
-	DateField,
-	TimestampField,
-	ColorField,
-	NumberField,
-	KeyTextField,
-	SelectField,
-	BooleanField,
-	GeoPointField,
-	// Complex
-	IntegrationFields,
-	GroupField,
-	Slice,
-	SharedSlice,
-	SharedSliceVariation,
-	SliceZone,
-	// Misc
-	AnyRegularField,
-	// Meta
-	FieldState,
-} from "./fields";
-
-// Deprecations (unused import for @link references)
-import type {
-	OEmbedExtra,
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	PhotoOEmbed,
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	VideoOEmbed,
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	LinkOEmbed,
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	RichOEmbed,
-	AnyOEmbed,
-} from "./fields";
-/**
- * @deprecated Use {@link PhotoOEmbed}, {@link VideoOEmbed}, {@link LinkOEmbed},
- *   {@link RichOEmbed}, or {@link AnyOEmbed} instead.
- */
-export type CommonEmbedData = AnyOEmbed & OEmbedExtra;
-
-export type {
-	CustomTypeModel,
-	CustomTypeModelDefinition,
-	CustomTypeModelTab,
-} from "./customType";
-
-export {
-	CustomTypeModelFieldType,
-	CustomTypeModelLinkSelectType,
-	CustomTypeModelSliceDisplay,
-	CustomTypeModelSliceType,
-} from "./customTypeModelFields";
-
-export type {
-	// Rich Text & Title
-	CustomTypeModelRichTextField,
-	CustomTypeModelTitleField,
-	CustomTypeModelRichTextMultiField,
-	CustomTypeModelRichTextSingleField,
-	// Image
-	CustomTypeModelImageField,
-	CustomTypeModelImageConstraint,
-	CustomTypeModelImageThumbnail,
-	// Links
-	CustomTypeModelContentRelationshipField,
-	CustomTypeModelLinkField,
-	CustomTypeModelLinkToMediaField,
-	// Embed
-	CustomTypeModelEmbedField,
-	// Simple
-	CustomTypeModelBooleanField,
-	CustomTypeModelColorField,
-	CustomTypeModelDateField,
-	CustomTypeModelKeyTextField,
-	CustomTypeModelNumberField,
-	CustomTypeModelSelectField,
-	CustomTypeModelTimestampField,
-	CustomTypeModelGeoPointField,
-	// Complex
-	CustomTypeModelGroupField,
-	CustomTypeModelIntegrationFieldsField,
-	// Slices
-	CustomTypeModelSliceZoneField,
-	CustomTypeModelSlice,
-	CustomTypeModelSliceLabel,
-	CustomTypeModelSharedSlice,
-	CustomTypeModelLegacySlice,
-	// SharedSlices
-	SharedSliceModel,
-	SharedSliceModelVariation,
-	// Depracated
-	CustomTypeModelRangeField,
-	CustomTypeModelSeparatorField,
-	// Misc
-	CustomTypeModelUIDField,
-	CustomTypeModelField,
-	CustomTypeModelFieldForGroup,
-} from "./customTypeModelFields";
-
-export type {
-	Repository,
-	Ref,
-	Release,
-	Language,
-	Query,
-	Form,
-	FormField,
-	Tags,
-} from "./api";
-
-export { WebhookType } from "./webhook";
-
-export type {
-	WebhookBody,
-	WebhookBodyAPIUpdate,
-	WebhookBodyTestTrigger,
-} from "./webhook";
+export * from "./api";
+export * from "./model";
+export * from "./value";
+export * from "./webhook";
```

Comment:
Up for discussion but I'm not a fan of having `export *` in the root index as it makes it harder to discover what's exported. Same for deprecations, to me it's simpler to have everything managed in one file and to keep the subfolder index files "clean" 🤔

What do you think?

Reply:
The root-level file will be quite large (as it is today), but I think we could make it work with some clear comments.

I'll adjust it and see which one we like better.

I think we should avoid having similar `src/index.ts` and `src/{api,model,…}/index.ts` files if we don't use `export *` statements. I'm in favor of removing the nested `index.ts` files in that case.